03-sql-joins-aliases.md

```diff
@@ -87,14 +87,40 @@ but using `AS` is much clearer so it's good style to include it.
 >
 > SQL queries help us *ask* specific *questions* which we want to answer about our data. The real skill with SQL is to know how to translate our scientific questions into a sensible SQL query (and subsequently visualize and interpret our results).
 >
-> Have a look at the following questions; these questions are written in plain English. Can you translate them to *SQL queries* and give a suitable answer?
-> - How many plots from each type are there?
-> - How many specimens are of each gender are there for each year?
-> - How many specimens of each species were captured in each type of plot?
-> - What is the average weight of each taxa?
-> - What is the percentage of specimens in each taxa?
-> - What are the minimum, maximum and average weight for each species of Rodent?
-> - What is the hindfoot length for male and female rodent of each species? Is there a Male / Female difference?
-> - What is the average weight of each rodent species over the course of the years? Is there any noticeable trend for any of the species?
+> Have a look at the following questions; these questions are written in plain English. Can you translate them to *SQL queries* and give a suitable answer?  
+> 1. How many plots from each type are there?  
+
+> 2. How many specimens are of each gender are there for each year?  
+
+> 3. How many specimens of each species were captured in each type of plot?  
+
+> 4. What is the average weight of each taxa?  
+
+> 5. What is the percentage of each species in each taxa?  
+
+> 6. What are the minimum, maximum and average weight for each species of Rodent?  
+
+> 7. What is the hindfoot length for male and female rodent of each species? Is there a Male / Female difference?  
+
+> 8. What is the average weight of each rodent species over the course of the years? Is there any noticeable trend for any of the species?  
+>
+> Proposed solutions:
+>
+> 1. Solution: ```SELECT plot_type, count(*) AS howmany  FROM plots  GROUP BY plot_type  ORDER BY howmany DESC```  
+
+> 2. Solution: ```SELECT year, sex, count(*) AS num_animal  FROM surveys  WHERE sex IS NOT null  GROUP BY sex, year```
+
+> 3. Solution: ```SELECT species_id, plot_type, count(*) FROM surveys JOIN plots USING (plot_id) WHERE species_id IS NOT null GROUP BY species_id, plot_type```
+
+> 4. Solution: ```SELECT taxa, AVG(weight) FROM species JOIN surveys USING(species_id) GROUP BY taxa```
+
+> 5. Solution: ```SELECT taxa, 100*count(*)/(SELECT cast(count(*) as float) FROM surveys) FROM surveys JOIN species USING (species_id ) GROUP BY taxa```
```

Comment:
CAST (I hate to say) might actually be worth considering as an extra challenge, given firstly that it's universally implemented and secondly that it's essential in certain use-cases (since SQL is a strongly and strictly typed language implying certain things simply can't be done without explicit type-casts). The above example is basic but I'd agree that simply using 100.0 is better in this case; perhaps an example casting strings to some numeric type would be useful?